Mastermind.pm

```diff
@@ -0,0 +1,229 @@
+
+  return \%header; 
+}
+
+sub run {
+  my ($self, $tva) = @_;
+    
+  my $vf = $tva->variation_feature; 
+  my $tv = $tva->transcript_variation; 
+  my $chr = $vf->{chr};
+
+  my $chr_syn;
+  my @new_chr_array;
+  my $new_chr;
+  
+  $self->parse_chromosome_synonyms($self->config->{'synonyms'}) if $self->config->{cache} && (not defined($self->{config}->{_chromosome_synonyms}));
+  
+  if(defined($self->{syn_cache}->{$chr})) {
+    $new_chr = $self->{syn_cache}->{$chr}; 
+  }
+  else {
+    if($self->config->{database}) {
+      my $srs_adaptor = $vf->slice->adaptor->db->get_SeqRegionSynonymAdaptor();
+      $chr_syn = $srs_adaptor->get_synonyms( $vf->slice->get_seq_region_id($vf->slice) );
+      @new_chr_array = map {$_->{name}} (grep {$_->{name} =~ 'NC_'} @{$chr_syn});
+    }
+    elsif($self->config->{cache}) {
+      $chr_syn = $self->config->{_chromosome_synonyms}->{($vf->{chr})};
+      @new_chr_array = grep(/NC_/, keys($chr_syn))
+    }
+
+    $new_chr = shift(@new_chr_array);
+    $self->{syn_cache}->{$chr} = $new_chr;
+  }
+  
+  my @alleles = split /\//, $vf->allele_string;  
+  my $ref_allele = shift @alleles; 
+  my $alt_allele = shift @alleles; 
+  
+  my $end = $vf->{end};
+  my $start = $vf->{start};
+  ($start, $end) = ($end, $start) if $start > $end;
+  my $consequence = $vf->consequence_type;  
+
+  my @data = @{$self->get_data($new_chr, $start, $end)}; 
+  
+  my $result_data = '';  
+  
+  foreach my $data_value (@data) {
+
+    if($data_value->{data}) {
+      my $ref_allele_comp = $ref_allele; 
+      my $alt_allele_comp = $alt_allele; 
+      reverse_comp(\$ref_allele_comp); 
+      reverse_comp(\$alt_allele_comp);  
+      
+      my $mm_ref = $data_value->{ref};
+      my $mm_alt = $data_value->{alt};
+
+      if( ($ref_allele eq $mm_ref && $alt_allele eq $mm_alt) || ($ref_allele_comp eq $mm_ref && $alt_allele_comp eq $mm_alt) ) { 
+        
+        my $peptide_start = defined($tv->translation_start) ? $tv->translation_start : undef;  
+        my $aa_alterations = $data_value->{aa};
+
+        foreach my $aa_alteration (@$aa_alterations) {
+          $aa_alteration =~ s/.*\:[A-Za-z]+//;
+          $aa_alteration =~ s/[A-Za-z]+|\*//;
+
+          # checks if citation refers to an UTR variant 
+          my $mm_utr = $aa_alteration =~/UTR/;
+          if($mm_utr) { 
+            $result_data = $data_value->{data};
+          }
+          # If there's a protein alteration then it only adds citations for the exact alteration cited  
```

Comment:
This is the exact location, rather than exact alteration, isn't it? This is fine, but it should be clear in the DESCRIPTION that this is how matching works.

Reply:
I'll explain that in the description. 
Thanks